vaultclient/vaultclient.go

```diff
@@ -1,63 +1,175 @@
-	"github.com/aws/aws-sdk-go/aws/client/metadata"
-	"github.com/aws/aws-sdk-go/aws/request"
-	"github.com/aws/aws-sdk-go/aws/signer/v4"
-	"github.com/aws/aws-sdk-go/private/protocol/query"
-	"github.com/aws/aws-sdk-go/private/protocol/restjson"
+	"context"
+	"crypto/sha256"
+	"encoding/hex"
+	"encoding/xml"
+	"fmt"
+	"io"
+	"net/http"
+	"net/url"
+	"strings"
+	"time"
+
+	"github.com/aws/aws-sdk-go-v2/aws"
+	v4 "github.com/aws/aws-sdk-go-v2/aws/signer/v4"
+	"github.com/aws/aws-sdk-go-v2/credentials"
+	"github.com/aws/aws-sdk-go-v2/service/iam/types"
 )
 
 // Vault provides the API operation methods for making requests to Vault.
 type Vault struct {
-	*client.Client
+	cfg      aws.Config
+	client   *http.Client
+	endpoint string
 }
 
 // Service information constants
 const (
-	ServiceName = "iam"       // Name of service.
-	EndpointsID = ServiceName // ID to lookup a service endpoint with.
-	ServiceID   = "Vault"     // ServiceID is a unique identifier of a specific service.
+	ServiceName = "iam" // Name of service.
 )
 
-// New creates a new instance of the Vault client with a session.
-func New(p client.ConfigProvider, cfgs ...*aws.Config) *Vault {
-	c := p.ClientConfig(EndpointsID, cfgs...)
-	return newClient(*c.Config, c.Handlers, c.PartitionID, c.Endpoint, c.SigningRegion, c.SigningName)
-}
-
-func newClient(cfg aws.Config, handlers request.Handlers, partitionID, endpoint, signingRegion, signingName string) *Vault {
-	svc := &Vault{
-		Client: client.New(
-			cfg,
-			metadata.ClientInfo{
-				ServiceName:   ServiceName,
-				ServiceID:     ServiceID,
-				SigningName:   signingName,
-				SigningRegion: signingRegion,
-				PartitionID:   partitionID,
-				Endpoint:      endpoint,
-				APIVersion:    "2010-05-08",
-			},
-			handlers,
-		),
+func New(accessKey, secretKey, sessionToken, endpoint, region string) *Vault {
+	return &Vault{
+		cfg: aws.Config{
+			Region: region,
+			Credentials: credentials.NewStaticCredentialsProvider(
+				accessKey,
+				secretKey,
+				sessionToken,
+			),
+		},
+		client:   &http.Client{},
+		endpoint: endpoint,
+	}
+}
+
+// makeAWSRequest is a helper function that handles AWS v4 signing and HTTP requests
+// This centralizes the common request logic used by all API operations
+func (c *Vault) makeAWSRequest(ctx context.Context, action string, formData url.Values) (*http.Response, error) {
+	formData.Set("Action", action)
+	formData.Set("Version", "2010-05-08")
+	endpoint := c.endpoint + "/"
+
+	requestBody := formData.Encode()
+	req, err := http.NewRequestWithContext(ctx, "POST", endpoint, strings.NewReader(requestBody))
+	if err != nil {
+		return nil, fmt.Errorf("failed to create request: %w", err)
+	}
+
+	req.Header.Set("Content-Type", "application/x-www-form-urlencoded")
+	req.Header.Set("Host", req.URL.Host)
+
+	// Hash the request body payload for AWS signing
+	hash := sha256.Sum256([]byte(requestBody))
+	payloadHash := hex.EncodeToString(hash[:])
+
+	// Sign the request using AWS SDK v2 signer
```

Comment:
```suggestion
	// Sign the request using AWS SDK signer
```

(`v2` is not really useful, and makes the comment look wrong since we use `v4` signer -from _auth v4_)